File: programmers/LEVEL-2/짝지어_제거하기_v1.py
from typing import Collection


def solution(s):
    temp = []
    for char in s:
        if temp and char == temp[-1]:
            temp.pop()
        else:
            temp.append(char)

    if temp: return 0
    else: return 1

print('answer: 1', solution('baabaa'))
print('answer: 0', solution('cdcd'))

File: programmers/LEVEL-2/짝지어_제거하기_v1.java
import java.util.Stack;

/**
 * LEVEL-2
 * 2017 팁스타운 > 짝지어 제거하기
 */
public class Solution {
    public static void main(String[] args) {
        System.out.println(solution("baabaa"));
    }

    public static int solution(String s)
    {
        Stack<Character> stack = new Stack<>();
        for (char ch : s.toCharArray()) {
            if (!stack.isEmpty() && stack.peek() == ch) {
                stack.pop();
            } else {
                stack.push(ch);
            }
        }

        return stack.isEmpty() ? 1 : 0;
    }
}